sha2/src/sha256.rs

```diff
@@ -40,8 +43,11 @@ cfg_if::cfg_if! {
 #[allow(dead_code)]
 fn to_u32s(block: &[u8; 64]) -> [u32; 16] {
     let mut res = [0u32; 16];
-    for (src, dst) in block.chunks_exact(4).zip(res.iter_mut()) {
-        *dst = u32::from_be_bytes(src.try_into().unwrap());
+    // note: we intentionally do not use `zip`-based code here since
+    // it results in a suboptimal codegen for `opt-level = "s"`
+    for i in 0..16 {
```

Comment:
This may be just an artifact of how godbolt (or the Rust compiler) generates assembly for `opt-level=s`, see: https://users.rust-lang.org/t/128919 But I will leave this change just to be extra safe.